File: VB/HyperLinkPropertyEditor.Win/Module.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating

Namespace HyperLinkPropertyEditor.Win
    Public NotInheritable Partial Class HyperLinkPropertyEditorWindowsFormsModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Protected Overrides Function GetDeclaredControllerTypes() As IEnumerable(Of Type)
            Return New Type() { GetType(GridListEditorOpenHyperLinkViewController) }
        End Function
        Protected Overrides Function GetRegularTypes() As IEnumerable(Of Type)
            Return New Type() { GetType(WinHyperLinkStringPropertyEditor) }
        End Function
        Public Overrides Function GetModuleUpdaters(ByVal objectSpace As IObjectSpace, ByVal versionFromDB As Version) As IEnumerable(Of ModuleUpdater)
            Return ModuleUpdater.EmptyModuleUpdaters
        End Function
        Protected Overrides Function GetDeclaredExportedTypes() As IEnumerable(Of Type)
            Return Type.EmptyTypes
        End Function
    End Class
End Namespace

File: CS/HyperLinkPropertyEditor.Win/Module.cs
using System;
using System.Collections.Generic;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;

namespace HyperLinkPropertyEditor.Win {
	public sealed partial class HyperLinkPropertyEditorWindowsFormsModule : ModuleBase {
		public HyperLinkPropertyEditorWindowsFormsModule() {
			InitializeComponent();
		}
		protected override IEnumerable<Type> GetDeclaredControllerTypes() {
			return new Type[] { typeof(GridListEditorOpenHyperLinkViewController) };
		}
		protected override IEnumerable<Type> GetRegularTypes() {
			return new Type[] { typeof(WinHyperLinkStringPropertyEditor) };
		}
		public override IEnumerable<ModuleUpdater> GetModuleUpdaters(IObjectSpace objectSpace, Version versionFromDB) {
			return ModuleUpdater.EmptyModuleUpdaters;
		}
		protected override IEnumerable<Type> GetDeclaredExportedTypes() {
			return Type.EmptyTypes;
		}
	}
}
